### PR TITLE
LPS-136753

### PR DESCRIPTION
**Pull request by dantewang**

### CODE REVIEW

**Comment by tinatian** [comment]
ci:test:relevant

**Comment by tinatian** [comment]
ci:test:sf

File: modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/persistence_impl.ftl [hunk 1]
```diff
@@ -371,6 +371,16 @@ public class ${entity.name}PersistenceImpl extends BasePersistenceImpl<${entity.
 	 */
 	@Override
 	public void cacheResult(List<${entity.name}> ${entity.pluralVariableName}) {
+		<#if serviceBuilder.isVersionGTE_7_0_0()>
+			if ((valueObjectFinderCacheListThreshold >= 0) &&
+				(${entity.pluralVariableName}.size() > valueObjectFinderCacheListThreshold)) {
+
+				clearCache(${entity.pluralVariableName});
```
Comment (tinatian) [inline modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/persistence_impl.ftl:378]: @dantewang You do not need to clear entity cache. The reason we clear finder cache as a different result may be stored in cache. But for entity cache, the entity is not changed as all the cacheResult() happens after a fetch which means no update to the entities.

File: portal-kernel/src/com/liferay/portal/kernel/service/persistence/impl/BasePersistenceImpl.java [hunk 2]
```diff
@@ -1038,6 +1038,9 @@ protected T updateImpl(T model) {
 	@Deprecated
 	protected boolean finderCacheEnabled = true;
 
+	protected int valueObjectFinderCacheListThreshold = GetterUtil.getInteger(
```
Comment (tinatian) [inline portal-kernel/src/com/liferay/portal/kernel/service/persistence/impl/BasePersistenceImpl.java:1041]: Hi @dantewang ,
After a second thought, I do not think we can add this here.
The reason is, a release version may use a updated service builder which contains the logic you added, but they may not have this backport to BasePersistenceImpl.
So let's move this to the concrete PersistenceImpl

Comment (dantewang) [reply portal-kernel/src/com/liferay/portal/kernel/service/persistence/impl/BasePersistenceImpl.java:1041]: @tinatian OK, got it.

**Pull request closed without merging**